### PR TITLE
ci: setup OSSF Scorecard and pin actions and Docker versions

### PR DESCRIPTION
**Pull request by mmorel-35**
### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by jakubdyszkiewicz** [comment]
do you know if this works with dependabot?
If dependabot sees `actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11` will it upgrade it?

**Comment by mmorel-35** [comment]
Absolutely ! It will even update the version in comment.

**Review by lahabana — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -31,3 +31,8 @@ updates:
     labels:
       - "dependencies"      
       - "ci/skip-test" # No need to run tests on github actions updates
+
+  - package-ecosystem: docker
+    directory: /tools/postgres
```
[the remark this answers is not in the document]
Comment (mmorel-35) [reply .github/dependabot.yml:36]: The Dockerfile is in /tools/postgres, that's why I configured the directory this way.

[the remark this answers is not in the document]
Comment (lahabana) [reply .github/dependabot.yml:36]: Right but line 19 there are /tools/releases it's seems to be unecessary duplication no?

[the remark this answers is not in the document]
Comment (mmorel-35) [reply .github/dependabot.yml:36]: I'm not sure dependabot is able to detect files in subfolder for Dockerfiles

**Pull request merged**